Injection of LibraryEvaluator seems to take alot in the main thread

**Describe the bug**

On initial startup and injection LibraryEvaluator seems cause ANR. Assuming getting the initial instance of LibraryEvaluator might be performance intensive and shouldn't be run in the main thread

**To Reproduce**
Steps to reproduce the behavior:
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error

**Expected behavior**
A clear and concise description of what you expected to happen.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Smartphone (please complete the following information):**
 - Device: [e.g. Samsung Galaxy A2 emulator]
 - Android version: [e.g. Settings -> About phone -> Android version]
 - Build number: [e.g. Settings -> About phone -> Build number]
 - Version [e.g. 24]

**Additional context**
Add any other context about the problem here.

```
com.github.anrwatchdog.ANRError: Application Not Responding for at least 5000 ms.
2023-10-16 13:00:26.241 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: main (state = RUNNABLE)
2023-10-16 13:00:26.242 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.android.icu.util.regex.PatternNative.compileImpl(Native Method)
2023-10-16 13:00:26.242 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.android.icu.util.regex.PatternNative.<init>(PatternNative.java:53)
2023-10-16 13:00:26.242 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.android.icu.util.regex.PatternNative.create(PatternNative.java:49)
2023-10-16 13:00:26.242 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.regex.Pattern.compile(Pattern.java:3533)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.regex.Pattern.<init>(Pattern.java:1419)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.regex.Pattern.compile(Pattern.java:921)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.regex.Pattern.matches(Pattern.java:1031)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.String.matches(String.java:2700)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.RuntimeResourceDefinition.massagePathForCompartmentSimilarity(RuntimeResourceDefinition.java:253)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.RuntimeResourceDefinition.sealAndInitialize(RuntimeResourceDefinition.java:210)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.ModelScanner.init(ModelScanner.java:157)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.ModelScanner.<init>(ModelScanner.java:95)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.FhirContext.scanResourceTypes(FhirContext.java:1000)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.FhirContext.validateInitialized(FhirContext.java:1113)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.FhirContext.getResourceDefinition(FhirContext.java:473)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.parser.ParserState$PreResourceState.enteringNewElement(ParserState.java:995)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.parser.ParserState.enteringNewElement(ParserState.java:137)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.parser.XmlParser.doXmlLoop(XmlParser.java:188)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.parser.XmlParser.parseResource(XmlParser.java:781)
2023-10-16 13:00:26.243 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.parser.XmlParser.doParseResource(XmlParser.java:147)
2023-10-16 13:00:26.246 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.parser.BaseParser.parseResource(BaseParser.java:601)
2023-10-16 13:00:26.246 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.parser.BaseParser.parseResource(BaseParser.java:654)
2023-10-16 13:00:26.246 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.support.DefaultProfileValidationSupport.parseBundle(DefaultProfileValidationSupport.java:360)
2023-10-16 13:00:26.246 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.support.DefaultProfileValidationSupport.loadStructureDefinitions(DefaultProfileValidationSupport.java:332)
2023-10-16 13:00:26.246 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.support.DefaultProfileValidationSupport.provideStructureDefinitionMap(DefaultProfileValidationSupport.java:254)
2023-10-16 13:00:26.246 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.support.DefaultProfileValidationSupport.fetchAllStructureDefinitions(DefaultProfileValidationSupport.java:159)
2023-10-16 13:00:26.246 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.hl7.fhir.common.hapi.validation.support.ValidationSupportChain.lambda$fetchAllStructureDefinitions$0(ValidationSupportChain.java:200)
2023-10-16 13:00:26.246 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.hl7.fhir.common.hapi.validation.support.ValidationSupportChain$$ExternalSyntheticLambda1.apply(Unknown Source:2)
2023-10-16 13:00:26.258 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.hl7.fhir.common.hapi.validation.support.ValidationSupportChain.doFetchStructureDefinitions(ValidationSupportChain.java:212)
2023-10-16 13:00:26.258 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.hl7.fhir.common.hapi.validation.support.ValidationSupportChain.fetchAllStructureDefinitions(ValidationSupportChain.java:200)
2023-10-16 13:00:26.258 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.hl7.fhir.r4.hapi.ctx.HapiWorkerContext.allStructures(HapiWorkerContext.java:77)
2023-10-16 13:00:26.258 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.hl7.fhir.r4.utils.FHIRPathEngine.<init>(FHIRPathEngine.java:263)
2023-10-16 13:00:26.258 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.hl7.fhir.r4.hapi.fluentpath.FhirPathR4.<init>(FhirPathR4.java:23)
2023-10-16 13:00:26.258 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.hl7.fhir.r4.hapi.ctx.FhirR4.createFhirPathExecutor(FhirR4.java:47)
2023-10-16 13:00:26.258 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ca.uhn.fhir.context.FhirContext.newFhirPath(FhirContext.java:805)
2023-10-16 13:00:26.258 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.opencds.cqf.cql.evaluator.fhir.util.FhirPathCache.lambda$cachedForContext$0(FhirPathCache.java:15)
2023-10-16 13:00:26.262 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.opencds.cqf.cql.evaluator.fhir.util.FhirPathCache$$ExternalSyntheticLambda0.apply(Unknown Source:4)
2023-10-16 13:00:26.262 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.HashMap.computeIfAbsent(HashMap.java:1221)
2023-10-16 13:00:26.262 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.opencds.cqf.cql.evaluator.fhir.util.FhirPathCache.cachedForContext(FhirPathCache.java:15)
2023-10-16 13:00:26.262 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.opencds.cqf.cql.evaluator.library.CqlFhirParametersConverter.<init>(CqlFhirParametersConverter.java:51)
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at **org.smartregister.fhircore.engine.cql.LibraryEvaluator.<init>(LibraryEvaluator.kt:81)**
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.smartregister.fhircore.quest.DaggerQuestApplication_HiltComponents_SingletonC$SingletonCImpl$SwitchingProvider.get(DaggerQuestApplication_HiltComponents_SingletonC.java:1459)
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.smartregister.fhircore.quest.DaggerQuestApplication_HiltComponents_SingletonC$ActivityCImpl.injectAppSettingActivity2(DaggerQuestApplication_HiltComponents_SingletonC.java:727)
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.smartregister.fhircore.quest.DaggerQuestApplication_HiltComponents_SingletonC$ActivityCImpl.injectAppSettingActivity(DaggerQuestApplication_HiltComponents_SingletonC.java:674)
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.smartregister.fhircore.engine.ui.appsetting.Hilt_AppSettingActivity.inject(Hilt_AppSettingActivity.java:69)
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at org.smartregister.fhircore.engine.ui.appsetting.Hilt_AppSettingActivity$1.onContextAvailable(Hilt_AppSettingActivity.java:40)
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at androidx.activity.contextaware.ContextAwareHelper.dispatchOnContextAvailable(ContextAwareHelper.kt:84)
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at androidx.activity.ComponentActivity.onCreate(ComponentActivity.java:377)
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:216)
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at **org.smartregister.fhircore.engine.ui.appsetting.AppSettingActivity.onCreate(AppSettingActivity.kt:52)**
2023-10-16 13:00:26.263 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.app.Activity.performCreate(Activity.java:8595)
2023-10-16 13:00:26.264 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.app.Activity.performCreate(Activity.java:8573)
2023-10-16 13:00:26.265 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1456)
2023-10-16 13:00:26.265 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3764)
2023-10-16 13:00:26.266 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3922)
2023-10-16 13:00:26.266 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
2023-10-16 13:00:26.266 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139)
2023-10-16 13:00:26.266 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96)
2023-10-16 13:00:26.266 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443)
2023-10-16 13:00:26.266 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.os.Handler.dispatchMessage(Handler.java:106)
2023-10-16 13:00:26.269 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.os.Looper.loopOnce(Looper.java:205)
2023-10-16 13:00:26.272 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.os.Looper.loop(Looper.java:294)
2023-10-16 13:00:26.273 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.app.ActivityThread.main(ActivityThread.java:8177)
2023-10-16 13:00:26.273 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.reflect.Method.invoke(Native Method)
2023-10-16 13:00:26.273 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
2023-10-16 13:00:26.273 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
2023-10-16 13:00:26.273 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: Crashlytics Exception Handler1 (state = WAITING)
2023-10-16 13:00:26.273 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at jdk.internal.misc.Unsafe.park(Native Method)
2023-10-16 13:00:26.282 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:341)
2023-10-16 13:00:26.282 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode.block(AbstractQueuedSynchronizer.java:506)
2023-10-16 13:00:26.282 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3466)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3437)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1623)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:435)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1071)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1131)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.google.firebase.crashlytics.internal.common.ExecutorUtils$1$1.onRun(ExecutorUtils.java:67)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.google.firebase.crashlytics.internal.common.BackgroundPriorityRunnable.run(BackgroundPriorityRunnable.java:27)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Thread.run(Thread.java:1012)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: FinalizerDaemon (state = WAITING)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Native Method)
2023-10-16 13:00:26.283 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Object.java:386)
2023-10-16 13:00:26.297 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:210)
2023-10-16 13:00:26.299 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:231)
2023-10-16 13:00:26.299 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:309)
2023-10-16 13:00:26.300 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Daemons$Daemon.run(Daemons.java:145)
2023-10-16 13:00:26.300 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 1 more
2023-10-16 13:00:26.300 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: FinalizerWatchdogDaemon (state = WAITING)
2023-10-16 13:00:26.300 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Native Method)
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Object.java:386)
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Object.java:524)
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Daemons$FinalizerWatchdogDaemon.sleepUntilNeeded(Daemons.java:428)
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Daemons$FinalizerWatchdogDaemon.runInternal(Daemons.java:408)
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 2 more
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: Firebase Background Thread #0 (state = WAITING)
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at jdk.internal.misc.Unsafe.park(Native Method)
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:341)
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode.block(AbstractQueuedSynchronizer.java:506)
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3466)
2023-10-16 13:00:26.301 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3437)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1623)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:435)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1071)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1131)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.google.firebase.concurrent.CustomThreadFactory.lambda$newThread$0$com-google-firebase-concurrent-CustomThreadFactory(CustomThreadFactory.java:47)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.google.firebase.concurrent.CustomThreadFactory$$ExternalSyntheticLambda0.run(Unknown Source:4)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 1 more
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: Firebase Blocking Thread #0 (state = TIMED_WAITING)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at jdk.internal.misc.Unsafe.park(Native Method)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:401)
2023-10-16 13:00:26.302 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:903)
2023-10-16 13:00:26.304 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1070)
2023-10-16 13:00:26.304 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 5 more
2023-10-16 13:00:26.304 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: Firebase Blocking Thread #1 (state = TIMED_WAITING)
2023-10-16 13:00:26.304 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 10 more
2023-10-16 13:00:26.304 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: Firebase Blocking Thread #2 (state = TIMED_WAITING)
2023-10-16 13:00:26.304 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 10 more
2023-10-16 13:00:26.304 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: Firebase Blocking Thread #3 (state = TIMED_WAITING)
2023-10-16 13:00:26.304 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 10 more
2023-10-16 13:00:26.304 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: Firebase Blocking Thread #4 (state = TIMED_WAITING)
2023-10-16 13:00:26.329 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 10 more
2023-10-16 13:00:26.329 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: Firebase Blocking Thread #5 (state = TIMED_WAITING)
2023-10-16 13:00:26.330 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 10 more
2023-10-16 13:00:26.330 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: Firebase Blocking Thread #6 (state = TIMED_WAITING)
2023-10-16 13:00:26.330 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 10 more
2023-10-16 13:00:26.330 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: GmsDynamite (state = WAITING)
2023-10-16 13:00:26.330 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Native Method)
2023-10-16 13:00:26.330 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Object.java:386)
2023-10-16 13:00:26.330 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Object.java:524)
2023-10-16 13:00:26.330 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.google.android.gms.dynamite.zza.run(com.google.android.gms:play-services-basement@@18.1.0:2)
2023-10-16 13:00:26.330 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: Measurement Worker (state = RUNNABLE)
2023-10-16 13:00:26.330 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.System.arraycopy(Native Method)
2023-10-16 13:00:26.331 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.content.pm.Signature.toByteArray(Signature.java:212)
2023-10-16 13:00:26.331 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at lx.f(:com.google.android.gms.dynamite_measurementdynamite@231818044@23.18.18 (190400-0):17)
2023-10-16 13:00:26.331 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at kk.r(:com.google.android.gms.dynamite_measurementdynamite@231818044@23.18.18 (190400-0):8)
2023-10-16 13:00:26.331 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at ix.run(:com.google.android.gms.dynamite_measurementdynamite@231818044@23.18.18 (190400-0):124)
2023-10-16 13:00:26.331 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
2023-10-16 13:00:26.331 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at iu.run(:com.google.android.gms.dynamite_measurementdynamite@231818044@23.18.18 (190400-0):5)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: ReferenceQueueDaemon (state = WAITING)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Native Method)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Object.java:386)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Object.wait(Object.java:524)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Daemons$ReferenceQueueDaemon.runInternal(Daemons.java:239)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Daemons$Daemon.run(Daemons.java:145)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Thread.run(Thread.java:1012)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: ScionFrontendApi (state = TIMED_WAITING)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at jdk.internal.misc.Unsafe.park(Native Method)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:1672)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.LinkedBlockingQueue.poll(LinkedBlockingQueue.java:460)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1070)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1131)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 1 more
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: awaitEvenIfOnMainThread task continuation executor1 (state = WAITING)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at jdk.internal.misc.Unsafe.park(Native Method)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:341)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode.block(AbstractQueuedSynchronizer.java:506)
2023-10-16 13:00:26.332 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3466)
2023-10-16 13:00:26.333 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3437)
2023-10-16 13:00:26.333 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1623)
2023-10-16 13:00:26.333 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:435)
2023-10-16 13:00:26.333 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1071)
2023-10-16 13:00:26.333 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1131)
2023-10-16 13:00:26.333 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
2023-10-16 13:00:26.333 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.google.firebase.crashlytics.internal.common.ExecutorUtils$1$1.onRun(ExecutorUtils.java:67)
2023-10-16 13:00:26.333 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.google.firebase.crashlytics.internal.common.BackgroundPriorityRunnable.run(BackgroundPriorityRunnable.java:27)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 1 more
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: com.google.firebase.crashlytics.startup1 (state = WAITING)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 13 more
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: pool-10-thread-1 (state = TIMED_WAITING)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at jdk.internal.misc.Unsafe.park(Native Method)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:252)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.awaitNanos(AbstractQueuedSynchronizer.java:1672)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.LinkedBlockingQueue.poll(LinkedBlockingQueue.java:460)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1070)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1131)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 1 more
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: pool-16-thread-1 (state = WAITING)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at jdk.internal.misc.Unsafe.park(Native Method)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:341)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionNode.block(AbstractQueuedSynchronizer.java:506)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ForkJoinPool.unmanagedBlock(ForkJoinPool.java:3466)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3437)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:1623)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:435)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1071)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 3 more
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: pool-4-thread-1 (state = WAITING)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 11 more
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: pool-8-thread-1 (state = WAITING)
2023-10-16 13:00:26.336 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	... 11 more
2023-10-16 13:00:26.337 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: queued-work-looper (state = RUNNABLE)
2023-10-16 13:00:26.338 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.os.MessageQueue.nativePollOnce(Native Method)
2023-10-16 13:00:26.338 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.os.MessageQueue.next(MessageQueue.java:335)
2023-10-16 13:00:26.338 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.os.Looper.loopOnce(Looper.java:162)
2023-10-16 13:00:26.338 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.os.Looper.loop(Looper.java:294)
2023-10-16 13:00:26.338 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at android.os.HandlerThread.run(HandlerThread.java:67)
2023-10-16 13:00:26.339 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  Caused by: com.github.anrwatchdog.ANRError$$$_Thread: |ANR-WatchDog| (state = RUNNABLE)
2023-10-16 13:00:26.339 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at dalvik.system.VMStack.getThreadStackTrace(Native Method)
2023-10-16 13:00:26.339 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Thread.getStackTrace(Thread.java:1841)
2023-10-16 13:00:26.339 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at java.lang.Thread.getAllStackTraces(Thread.java:1909)
2023-10-16 13:00:26.349 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.github.anrwatchdog.ANRError.New(ANRError.java:77)
2023-10-16 13:00:26.349 11518-11567 QuestApplication        org...rtregister.fhircore.mwcoreDev  E  	at com.github.anrwatchdog.ANRWatchDog.run(ANRWatchDog.java:273)

```

--- a/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/appsetting/AppSettingActivity.kt
+++ b/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/appsetting/AppSettingActivity.kt
@@ -30,6 +30,7 @@ import androidx.compose.ui.res.stringResource
 import androidx.lifecycle.lifecycleScope
 import dagger.hilt.android.AndroidEntryPoint
 import javax.inject.Inject
+import javax.inject.Provider
 import kotlinx.coroutines.launch
 import org.smartregister.fhircore.engine.R
 import org.smartregister.fhircore.engine.cql.LibraryEvaluator
@@ -51,7 +52,7 @@ class AppSettingActivity : AppCompatActivity() {
 
   @Inject lateinit var dispatcherProvider: DispatcherProvider
 
-  @Inject lateinit var libraryEvaluator: LibraryEvaluator
+  @Inject lateinit var libraryEvaluatorProvider: Provider<LibraryEvaluator>
   val appSettingViewModel: AppSettingViewModel by viewModels()
 
   override fun onCreate(savedInstanceState: Bundle?) {
@@ -68,7 +69,7 @@ class AppSettingActivity : AppCompatActivity() {
         }
       }
     }
-    lifecycleScope.launch(dispatcherProvider.io()) { libraryEvaluator.initialize() }
+    lifecycleScope.launch(dispatcherProvider.io()) { libraryEvaluatorProvider.get().initialize() }
     val existingAppId =
       sharedPreferencesHelper.read(SharedPreferenceKey.APP_ID.name, null)?.trimEnd()
 

--- a/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt
+++ b/android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt
@@ -34,6 +34,7 @@ import dagger.hilt.android.lifecycle.HiltViewModel
 import java.util.Date
 import java.util.UUID
 import javax.inject.Inject
+import javax.inject.Provider
 import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.withContext
@@ -92,7 +93,7 @@ constructor(
   val resourceDataRulesExecutor: ResourceDataRulesExecutor,
   val transformSupportServices: TransformSupportServices,
   val sharedPreferencesHelper: SharedPreferencesHelper,
-  val libraryEvaluator: LibraryEvaluator,
+  val libraryEvaluatorProvider: Provider<LibraryEvaluator>,
   val fhirPathDataExtractor: FhirPathDataExtractor,
 ) : ViewModel() {
 
@@ -601,7 +602,7 @@ constructor(
     }
     questionnaire.cqfLibraryIds().forEach { libraryId ->
       if (subject.resourceType == ResourceType.Patient) {
-        libraryEvaluator.runCqlLibrary(libraryId, subject as Patient, bundle)
+        libraryEvaluatorProvider.get().runCqlLibrary(libraryId, subject as Patient, bundle)
       }
     }
   }

--- a/android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt
+++ b/android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt
@@ -175,7 +175,7 @@ class QuestionnaireViewModelTest : RobolectricTest() {
           transformSupportServices = mockk(),
           dispatcherProvider = defaultRepository.dispatcherProvider,
           sharedPreferencesHelper = sharedPreferencesHelper,
-          libraryEvaluator = libraryEvaluator,
+          libraryEvaluatorProvider = { libraryEvaluator },
           fhirCarePlanGenerator = fhirCarePlanGenerator,
           resourceDataRulesExecutor = resourceDataRulesExecutor,
           fhirPathDataExtractor = fhirPathDataExtractor,

